### ISSUE TITLE
ToFormData - Which way should we proceed? 

### ISSUE THREAD

**Issue by carpben** [axios/axios#4471]
https://github.com/axios/axios/pull/3757 added a function converts a JS object to a FormData format. 
My suggestion was, to export it independently for now, and proceed in a risk free approach. 
@DigitalBrainJS thought we should add it by default if data is not in FormData format, and content-type is `multipart/form-data`. @DigitalBrainJS merged https://github.com/axios/axios/pull/4413 accordingly. 

Recent `FormData` issues (such as https://github.com/axios/axios/issues/4406) show that tests related to `FormData` are lacking. . If we do decide to publish https://github.com/axios/axios/pull/4413 we need to make sure that there are proper tests for using `toFormData` by default. More specifically, that current usage, passing data in `FormData` format, is not affected at all by the change. 

@jasonsaayman 
#### Additional context

**Comment by jasonsaayman** [comment]
Can you please add this to the discussion :)

**Comment by carpben** [comment]
@DigitalBrainJS I did not miss that point (see quote at the beginning of your comment 🙂). 
Your claim that currently JS object are serialized to JSON is a strong one, and it makes sense to be consistent. 

I thought we need to have a test if a user passes FormData current change doesn't affect it in any way. But I noticed a test you added in https://github.com/axios/axios/pull/4448 that solves this issue. 

I guess my only reservation currently is that at first we should not recommend it as the preferred method of choice in the documentation. Be more gradual. First see how it is accepted (No major issues/bugs) and only then recommend it as the first/default method.
